lib/plugins/input/docker/dockerInspect.js

```diff
@@ -31,6 +30,25 @@ docker.info(function dockerInfoHandler (err, data) {
   }
 })
 
+function parseReceiverList (receivers) {
+  let rv = []
+  if (!receivers) {
+    return undefined
+  }
+  let receiverList = receivers.split(',')
+  let parseUrlRegx = /(\S+:\/\/\S+?)\/(\S+)$/i
+  for (let i = 0; i < receiverList.length; i++) {
```

Comment:
Can you refactor this to use higher-order functions or iterators? For any new code we add to Node.js projects, I think we should focus on writing functional code. This will reduce state mutations and be much less prone to bugs, but also much easier to test and review.

Reply:
C was not your first programming language, right? Done :)